compiler/rustc_symbol_mangling/src/v0.rs

```diff
@@ -602,26 +630,9 @@ impl<'tcx> Printer<'tcx> for SymbolMangler<'tcx> {
                 });
 
                 match inner_ty.kind() {
-                    ty::Str if mutbl.is_not() => {
-                        let slice =
-                            valtree.try_to_raw_bytes(self.tcx(), ct_ty).unwrap_or_else(|| {
-                                bug!(
-                                    "expected to get raw bytes from valtree {:?} for type {:}",
-                                    valtree,
-                                    ct_ty
-                                )
-                            });
-                        let s =
-                            std::str::from_utf8(slice).expect("non utf8 str from MIR interpreter");
-
-                        self.push("e");
-
-                        // FIXME(eddyb) use a specialized hex-encoding loop.
-                        for byte in s.bytes() {
-                            let _ = write!(self.out, "{byte:02x}");
-                        }
-
-                        self.push("_");
+                    ty::Str => {
+                        assert!(mutbl.is_not(), "&mut str is not expected");
+                        print_str(self);
```

Comment:
Is this match arm actually necessary? I think it'd be nicer if `print_str` was just inlined into the top level `ty::Str` match arm, and we removed this match statement altogether, always having the behaviour of the wildcard pattern

Reply:
Yes, it would be nicer if we handle `ty::Str` once and for all at the top level.

One remain problem is that we will then create a `Const` with a `ty::Str` type. I am not sure if this is okay for the spec/semantics of `Const`.

Reply:
It should be okay, if that's the only reason for not doing it that way then I think it would be fine to just go ahead with it.